dll/win32/shell32/shlexec.cpp

```diff
@@ -1863,7 +1863,7 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
         if(len-1 >= dwApplicationNameLen)
```

Comment:
Here, `len == length of (sei_tmp.lpFile + 1)` (it would have been clearer if code used:
`len = strlenW(sei_tmp.lpFile + 1)`.
Now, it may happen that len is "much larger" than  dwApplicationNameLen (that caps at MAX_PATH + 2).
If so, dwApplicationNameLen is adjusted. Otherwise, len is less than dwApplicationNameLen.
So, the allocation below should be enough to contain what's being copied.